### PR TITLE
[System] Add corefx tests for System.CodeDom

### PR DESCRIPTION
**Pull request by EgorBo**
Also fix [SkipOnTarget(NetFramework)] behavior.
Mono should skip tests marked with this attribute - https://gist.github.com/EgorBo/56c89420d29efacb330a7d14779cd4ce (currently it skips the first one)

### CODE REVIEW

**Comment by EgorBo** [comment]
there are some failures because of https://github.com/dotnet/buildtools/pull/1690

File: mcs/class/System/Makefile [hunk 1]
```diff
@@ -118,6 +119,7 @@ endif
 
 endif
 
+
```
Comment (akoeplinger) [inline mcs/class/System/Makefile]: extra line

File: mcs/class/System/System_xtest.dll.sources [hunk 2]
```diff
@@ -0,0 +1,19 @@
+../../../external/corefx/src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs
+../../../external/corefx/src/Common/tests/System/PlatformDetection.cs
+
+../../../external/corefx/src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs
+../../../external/corefx/src/Common/tests/System/Collections/IEnumerable.NonGeneric.Tests.cs
+../../../external/corefx/src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs
+../../../external/corefx/src/Common/tests/System/Collections/TestBase.NonGeneric.cs
+../../../external/corefx/src/common/tests/system/runtime/serialization/formatters/binaryformatterhelpers.cs
```
Comment (akoeplinger) [inline mcs/class/System/System_xtest.dll.sources]: the casing of the path looks wrong here (common vs Common) and it'll fail on Linux.

File: scripts/ci/run-test-default.sh [hunk 3]
```diff
@@ -58,6 +58,7 @@ ${TESTCMD} --label=System.Configuration --timeout=5m make -w -C mcs/class/System
 ${TESTCMD} --label=System.Transactions --timeout=5m make -w -C mcs/class/System.Transactions run-test
 ${TESTCMD} --label=System.Web.Extensions --timeout=5m make -w -C mcs/class/System.Web.Extensions run-test
 ${TESTCMD} --label=System.Core --timeout=15m make -w -C mcs/class/System.Core run-test
+${TESTCMD} --label=System --timeout=5m make -w -C mcs/class/System run-xunit-test
```
Comment (akoeplinger) [inline scripts/ci/run-test-default.sh]: change it to --label=System-xunit please. also move it up to the System run, it's below System.Core right now :)

File: mcs/class/System/System_xtest.dll.sources (hunk 2 above)
Comment (EgorBo) [reply mcs/class/System/System_xtest.dll.sources]: oh yeah it's not the first tim I hit this thing - the list of files where extracted from corefx' crprojs 😞  thanks for noticing that, created this PR on windows

Comment (akoeplinger) [reply mcs/class/System/System_xtest.dll.sources]: interesting. this should fail in CoreFx too then I think? anyway, please send an upstream PR to align the csproj path casing with actual disk layout :)

Comment (EgorBo) [reply mcs/class/System/System_xtest.dll.sources]: I guess MSBuild handles that ;-) but I'll create a PR in the corefx to cleanup such things

**Comment by EgorBo** [comment]
@monojenkins build

**Comment by EgorBo** [comment]
Blocked by https://github.com/dotnet/corefx/pull/25113 (two tests fail)

**Comment by marek-safar** [comment]
build

**Comment by marek-safar** [comment]
@EgorBo could you finish this PR

**Comment by EgorBo** [comment]
@marek-safar should be ready to merge, I had to ignore few tests but will bring them back later (I have a new set of SkipOnMono for the upstream).

**Comment by EgorBo** [comment]
oops. accidentally closed it. Reopening...

**Pull request merged**